openverse_catalog/dags/common/loader/reporting.py

```diff
@@ -72,19 +73,28 @@ def clean_duration(duration: float | list[float]):
 def clean_record_counts(
     record_counts_by_media_type: MediaTypeRecordMetrics | list[MediaTypeRecordMetrics],
     media_types: Sequence[str],
-):
+) -> dict[str, RecordMetrics]:
     # If a list of record_counts dicts is provided, sum all of the individual values
     if isinstance(record_counts_by_media_type, list):
         return {
             media_type: sum(
-                [x[media_type] for x in record_counts_by_media_type],
+                (x[media_type] for x in record_counts_by_media_type),
                 RecordMetrics(0, 0, 0, 0),
             )
             for media_type in media_types
         }
     return record_counts_by_media_type
 
 
+def skip_report_completion(
+    duration: str | None,
+    record_counts_by_media_type: dict[str, RecordMetrics],
+) -> bool:
+    return (duration is None or duration in ("inf", "less than 1 sec")) and (
+        not record_counts_by_media_type
```

Comment:
Slight complication here, it looks like `record_counts_by_media_type` will always (if not at least frequently) be a non-empty dictionary 😕 I ran the SMK DAG locally with the following diff: 

```diff
diff --git a/openverse_catalog/dags/common/loader/reporting.py b/openverse_catalog/dags/common/loader/reporting.py
index aefc662..f7dbd8a 100644
--- a/openverse_catalog/dags/common/loader/reporting.py
+++ b/openverse_catalog/dags/common/loader/reporting.py
@@ -90,6 +90,8 @@ def skip_report_completion(
     duration: str | None,
     record_counts_by_media_type: dict[str, RecordMetrics],
 ) -> bool:
+    print(f"{duration=}")
+    print(f"{record_counts_by_media_type=}")
     return (duration is None or duration in ("inf", "less than 1 sec")) and (
         not record_counts_by_media_type
     )
diff --git a/openverse_catalog/dags/providers/factory_utils.py b/openverse_catalog/dags/providers/factory_utils.py
index ca66e51..7f92b77 100644
--- a/openverse_catalog/dags/providers/factory_utils.py
+++ b/openverse_catalog/dags/providers/factory_utils.py
@@ -51,6 +51,7 @@ def pull_media_wrapper(
     # Not passing args or kwargs here because Airflow throws a bunch of stuff in
     # there that none of our provider scripts are expecting.
     try:
+        raise ValueError("Whoops!")
         data = ingester.ingest_records()
     finally:
         end_time = time.perf_counter()
```

Here were the logs:
```
[2023-03-24, 22:51:53 UTC] {logging_mixin.py:137} INFO - duration='less than 1 sec'
[2023-03-24, 22:51:53 UTC] {logging_mixin.py:137} INFO - record_counts_by_media_type={'image': None}
[2023-03-24, 22:51:53 UTC] {slack.py:321} INFO - 
*DAG*: `smk_workflow`
*Date range*: _all_
*Duration of data pull tasks*: less than 1 sec
*Number of records upserted per media type*:
  - `image`: _No data_
```

So it looks like our check here will need to be twofold:

1. Check that `record_counts_by_media_type` isn't falsy (which is what we have)
2. Check that _all dictionary values are not `None`_ (the new check we need to add)

This will also require some new test tweaks/additions as well.